src/snowflake/snowpark/_internal/analyzer/schema_utils.py

```diff
@@ -2,21 +2,27 @@
 # Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
 #
 
-from typing import TYPE_CHECKING, List
+from typing import TYPE_CHECKING, List, Union
 
 import snowflake.snowpark
-from snowflake.connector.constants import FIELD_ID_TO_NAME
-from snowflake.connector.cursor import ResultMetadata
+from snowflake.connector.cursor import ResultMetadata, SnowflakeCursor
 from snowflake.snowpark._internal.analyzer.analyzer_utils import (
     quote_name_without_upper_casing,
 )
 from snowflake.snowpark._internal.analyzer.expression import Attribute
-from snowflake.snowpark._internal.type_utils import convert_sf_to_sp_type
+from snowflake.snowpark._internal.type_utils import convert_metadata_to_sp_type
 from snowflake.snowpark.types import DecimalType, LongType, StringType
 
 if TYPE_CHECKING:
     import snowflake.snowpark.session
 
+    # Refer to the docstring for ResultMetadataV2 (in cursor.py) for more information about
+    # how it differs from ResultMetadata
```

Comment:
We could also bump the connector dependency lower bound to make sure `ResultMetadataV2` exists. Also, make sure sp connector has the same change released.

Reply:
Is this safe to do? That would definitely clean up the code, but I wasn't sure whether we could bump the version since this change is going out next week with the corresponding connector version that it would rely on. If that's fine to do, I can make the update

Reply:
We typically first release connector, then snowpark, so it should be safe. We also bumped the version for geometry type I think. cc @sfc-gh-aalam to confirm

Reply:
Will the merge gates pull a version of the connector from main? If not, I don't think they'll pass once I make the update.

Reply:
To my understanding, only vector related function will be affected if older version of connector is used, right? We would need to bump connector dependency when we want to broader vector support release like this: https://github.com/snowflakedb/snowpark-python/pull/1156/files

Reply:
Yeah, if an older connector is used, vector queries would fail but everything else should be fine (iiuc, this is verified as the CI gates are running the currently-released connector and are passing). I think we'll likely want to bump the connector dependency when we exit PrPr.